src/nbsap/forms.py

```diff
@@ -781,15 +799,18 @@ def __init__(self, *args, **kwargs):
                 EuTarget.objects.exclude(id__in=existing_strategies).all(),
                 ['pk']
             )
+            if len(self.fields['eu_targets'].choices) == 0:
+                self.fields['eu_targets'].overlay = "No available targets"
 
     def clean(self):
         cleaned_data = super(EuAichiStrategyForm, self).clean()
         aichi_targets = cleaned_data.get('aichi_targets', [])
         other_aichi_targets = cleaned_data.get('other_aichi_targets', [])
         eu_indicators = cleaned_data.get('eu_indicators', [])
         other_eu_indicators = cleaned_data.get('other_eu_indicators', [])
+        aichi_goals = cleaned_data.get('aichi_goals', [])
 
-        if set(aichi_targets) & set(other_aichi_targets):
+        if set(aichi_targets) & set(other_aichi_targets) & set(aichi_goals):
```

Comment:
Why would a intersection between target ids and goal ids be a problem? These are different data types.